src/components/SwapWidget/SwapWidget.tsx

```diff
@@ -0,0 +1,247 @@
+            dispatch(
+              request({
+                chainId: chainId!,
+                senderToken: senderToken!,
+                senderAmount,
+                signerToken: signerToken!,
+                senderWallet: account!,
+                provider: library,
+              })
+            );
+            trackEvent({ category: "order", action: "request" });
+          }}
+        >
+          {!insufficientBalance ? 
+            (!senderAmount || parseFloat(senderAmount) === 0) ? 
+            t("orders:enterAmount") : t("orders:request") : 
+            t("orders:insufficentBalance", {
+              symbol: findTokenByAddress(senderToken!, activeTokens).symbol,
+            })
+          }
+        </Button>
+      )
+    }
+  }
+
+  let signerAmount: string | null = null;
+  if (order) {
+      signerAmount = toDecimalString(
+                      order.signerAmount, 
+                      getTokenDecimals(order.signerToken)
+                    );
+  }
+
+  let parsedSenderAmount = null;
+  let insufficientBalance: boolean = false;
+  if (senderAmount && senderToken && balances) {
+    if (parseFloat(senderAmount) === 0) {
+      insufficientBalance = false;
+    } else {
+      const decimals = getTokenDecimals(senderToken);
+      parsedSenderAmount = parseUnits(senderAmount, decimals);
+      insufficientBalance = BigNumber.from(
+        balances.values[senderToken!] || toAtomicString("0", 18)
+      ).lt(parsedSenderAmount);
+    }
+  }
+
+  return (
+    <Card className="flex-col m-4 w-72">
+      {!order ? 
+        <h3 className="mb-4 font-bold">Swap now</h3> : 
+        (<p className="mb-4">
+            Quote expires in&nbsp;
+            <Timer
+              expiryTime={parseInt(order.expiry)}
+              onTimerComplete={() => {
+                dispatch(
+                  request({
+                    chainId: chainId!,
+                    senderToken: senderToken!,
+                    senderAmount,
+                    signerToken: signerToken!,
+                    senderWallet: account!,
+                    provider: library,
+                  })
+                );
+                trackEvent({ category: "order", action: "request" });
+              }}
+            />
+        </p>)
+      }
+      <TokenSelect
+        tokens={activeTokens}
+        withAmount={true}
+        amount={senderAmount}
+        onAmountChange={(e) => handleTokenAmountChange(e)}
+        className="mb-2"
+        label={t("orders:send")}
+        token={senderToken}
+        onTokenChange={(e) => setSenderToken(e.currentTarget.value)}
+        hasError={insufficientBalance}
+      />
+      <TokenSelect
+        tokens={activeTokens}
+        withAmount={false}
+        className="mb-2"
+        label={t("orders:receive")}
+        token={signerToken}
+        quoteAmount={signerAmount}
+        onTokenChange={(e) => setSignerToken(e.currentTarget.value)}
+      />
+      {DisplayedButton()}
```

Comment:
Rather write as a JSX element rather than a function call? <DisplayedButton />

Reply:
Yeah true, will fix in the next commit.